src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -225,6 +226,34 @@ public override void OnCustom(string name, Activity activity, object payload)
             }
         }
 
+        public override void OnException(Activity activity, object payload)
+        {
+            if (activity.IsAllDataRequested)
+            {
+                if (!this.stopExceptionFetcher.TryFetch(payload, out Exception exc) || exc == null)
+                {
+                    AspNetCoreInstrumentationEventSource.Log.NullPayload(nameof(HttpInListener), nameof(this.OnException));
+                    return;
+                }
+
+                try
+                {
+                    this.options.Enrich?.Invoke(activity, "OnException", exc);
+                }
+                catch (Exception ex)
+                {
+                    AspNetCoreInstrumentationEventSource.Log.EnrichmentException(ex);
+                }
+
+                if (this.options.RecordException)
```

Comment:
We should probably do a follow-up PR to add this to AspNet instrumentation too since they are very similar?

Reply:
Need to if aspnet publishes an event with Exception in payload. If yes, we should add a follow up pR.

Reply:
I checked, it doesn't. My bad! 🤦